centreon/www/install/php/Update-next.php

```diff
@@ -33,7 +33,51 @@
  * @var ConnectionInterface $pearDBO
  */
 
-// TODO add your functions here
+/** -------------------------------------- Host Group Topology -------------------------------------- */
+$fixDuplicateHostGroupTopology = function () use ($pearDB, &$errorMessage, $version): void {
+    $errorMessage = 'Unable to fix duplicate Host Groups topology';
+    CentreonLog::create()->info(
+        logTypeId: CentreonLog::TYPE_UPGRADE,
+        message: "UPGRADE - {$version}: [topology] Fixing duplicate Host Groups menu entries",
+    );
+
+    // update ACL menu giving access to old topology 60102 to new topology 60105
+    $pearDB->update(
+        <<<'SQL'
+            UPDATE `acl_topology_relations` AS atr
+            LEFT JOIN `acl_topology_relations` AS atr2
+                ON atr2.`acl_topo_id` = atr.`acl_topo_id`
+                AND atr2.`topology_topology_id` = (SELECT `topology_id` FROM `topology` WHERE `topology_page` = 60105 LIMIT 1)
+            SET atr.`topology_topology_id` = (SELECT `topology_id` FROM `topology` WHERE `topology_page` = 60105 LIMIT 1)
+            WHERE atr.`topology_topology_id` = (SELECT `topology_id` FROM `topology` WHERE `topology_page` = 60102 LIMIT 1)
+                AND EXISTS (SELECT 1 FROM `topology` WHERE `topology_page` = 60102)
+                AND EXISTS (SELECT 1 FROM `topology` WHERE `topology_page` = 60105)
+                AND atr2.`acl_topo_id` IS NULL
+            SQL
+    );
+
+    // delete ACL reference to old topology 60102
+    $pearDB->delete(
+        <<<'SQL'
+            DELETE FROM `acl_topology_relations`
+            WHERE `topology_topology_id` = (SELECT `topology_id` FROM `topology` WHERE `topology_page` = 60102 LIMIT 1)
+                AND EXISTS (SELECT 1 FROM `topology` WHERE `topology_page` = 60102)
+            SQL
+    );
+
+    // make sure any topology entry 60102 is gone
+    $pearDB->delete(
+        <<<'SQL'
+            DELETE FROM `topology`
+            WHERE `topology_page` = 60102
+            SQL
+    );
```

Comment:
Are you sure about that ?
the acl_topology_relation have an on delete cascade set on the FK topology_topology_id referencing topology.topology_id

Reply:
I think it might be simpler to make sure that the original topology 60102 is sufficient for the react page to function and delete the 60105 one